master/internal/api_trials.go

```diff
@@ -419,3 +409,141 @@ func (a *apiServer) GetTrial(_ context.Context, req *apiv1.GetTrialRequest) (
 
 	return resp, nil
 }
+
+func (a *apiServer) GetTrialProfilerMetrics(
+	req *apiv1.GetTrialProfilerMetricsRequest,
+	resp apiv1.Determined_GetTrialProfilerMetricsServer,
+) error {
+	switch exists, err := a.m.db.CheckTrialExists(int(req.Labels.TrialId)); {
+	case err != nil:
+		return err
+	case !exists:
+		return status.Error(codes.NotFound, "trial not found")
+	}
+
+	labelsParam, err := protojson.Marshal(req.Labels)
+	if err != nil {
+		return fmt.Errorf("failed to marshal labels: %w", err)
+	}
+
+	fetch := func(lr api.LogsRequest) (api.LogBatch, error) {
+		switch {
+		case lr.Follow, lr.Limit > trialProfilerMetricsBatchSize:
+			lr.Limit = trialProfilerMetricsBatchSize
+		case lr.Limit <= 0:
+			return nil, nil
+		}
+
+		var batchOfBatches []*trialv1.TrialProfilerMetricsBatch
+		return model.TrialProfilerMetricsBatchBatch(batchOfBatches), a.m.db.QueryProto(
+			"get_trial_profiler_metrics",
+			&batchOfBatches, labelsParam, lr.Offset, lr.Limit,
+		)
+	}
+
+	onBatch := func(b api.LogBatch) error {
+		return b.ForEach(func(r interface{}) error {
+			return resp.Send(&apiv1.GetTrialProfilerMetricsResponse{
+				Batch: r.(*trialv1.TrialProfilerMetricsBatch),
+			})
+		})
+	}
+
+	return a.m.system.MustActorOf(
+		actor.Addr("trialMetricsStream-"+uuid.New().String()),
+		api.NewLogStoreProcessor(
```

Comment:
There is a follow up where i just use goland to rename and move a bunch of this "log" crap. Just putting it in the same PR makes it unreviewable.